lib/spack/spack/audit.py

```diff
@@ -265,6 +265,14 @@ def _search_duplicate_specs_in_externals(error_cls):
     kwargs=('pkgs',)
 )
 
+#: Sanity checks on package directives
+package_github_patch_url = AuditClass(
+    group='packages',
+    tag='PKG-GITHUB-PATCH-URL',
+    description='Enforce use of ?full_index=1 for GitHub patch URLs',
+    kwargs=('pkgs',)
+)
```

Comment:
Is there a reason not to reuse `package_directives` since we are auditing content in a directive? I think it makes sense to run these tests together with the tests on `depends_on` etc.

Reply:
Do you want the tags to be coarse or fine?  It seems useful to me to have specific tags, ala `flake8` with `E501`, etc.  But I don't really care -- I can refractor this one if you want.

Reply:
this is refactored.